viewing_party/main.py

```diff
@@ -0,0 +1,115 @@
+#WAVE 1 *******************************************************************************************************************************************************
+
+def create_movie(title, genre, rating):
+    while title and genre and rating:
+        return {"title":title, "genre":genre, "rating":rating}
+        
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for header in user_data["watchlist"]:
+        if header["title"] == title:
+            user_data["watchlist"].remove(header)
+            user_data["watched"].append(header)
+    return user_data
+
+
+#WAVE 2 *******************************************************************************************************************************************************
+
+def get_watched_avg_rating(user_data):
+    total = 0
+    if len(user_data["watched"]) == 0:
+        return 0
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    else:
+        return total/len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    all_genres = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] not in all_genres:
+            all_genres[movie["genre"]] = 1
+        else:
+            all_genres[movie["genre"]] += 1
+    if len(user_data["watched"]) == 0:
+        return None
+    most_watched = max(all_genres, key=all_genres.get)
+    return most_watched
```

Comment:
one thing to think about is this for loop is going to run whether or not there's anything in the `watched` list. start with the `if len(user_data["watched"])` first, that way the lines of code will run ONLY when needed